### PR TITLE
add inception python example

### PR DESCRIPTION
**Pull request by jenniew**
## What changes were proposed in this pull request?

add inception training example in python

## How was this patch tested?

manually tested

### CODE REVIEW

File: pyspark/bigdl/models/inception/README.md [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+## Train the Model
+* Spark standalone, example command
+```
+BigDL_HOME=...
+SPARK_HOME=...
+PYTHON_API_PATH=${BigDL_HOME}/dist/lib/bigdl-VERSION-python-api.zip
+BigDL_JAR_PATH=${BigDL_HOME}/dist/lib/bigdl-VERSION-jar-with-dependencies.jar
+PYTHONPATH=${PYTHON_API_PATH}:$PYTHONPATH
+source ${BigDL_HOME}/dist/bin/bigdl.sh
```
Comment (SeaOfOcean) [inline pyspark/bigdl/models/inception/README.md]: remove this

File: pyspark/bigdl/models/inception/inception.py [hunk 2]
```diff
@@ -0,0 +1,313 @@
...
+    model.add(SpatialCrossMapLRN(5, 0.0001, 0.75).set_name("pool1/norm1"))
+    model.add(SpatialConvolution(64, 64, 1, 1, 1, 1).set_init_method(weight_init_method=Xavier())
+              .set_name("conv2/3x3_reduce"))
+    model.add(ReLU(True).set_name("conv2/relu_3x3_reduce"))
+    model.add(SpatialConvolution(64, 192, 3, 3, 1, 1, 1, 1).set_init_method(weight_init_method=Xavier())
+              .set_name("conv2/3x3"))
+    model.add(ReLU(True).set_name("conv2/relu_3x3"))
+    model.add(SpatialCrossMapLRN(5, 0.0001, 0.75).set_name("conv2/norm2"))
+    model.add(SpatialMaxPooling(3, 3, 2, 2, to_ceil=True).set_name("pool2/3x3_s2"))
+    model.add(Inception_Layer_v1(192, scala_T([scala_T([64]), scala_T(
```
Comment (SeaOfOcean) [inline pyspark/bigdl/models/inception/inception.py]: why we need scala_T? it looks complicated

Comment (jenniew) [reply pyspark/bigdl/models/inception/inception.py]: it mimic table in scala. In python, we could use python list directly, but the index comes from 0. To be consistent with scala model, create this structure. And I changed the name from scala_T to T.

File: pyspark/bigdl/models/inception/README.md (hunk 1 above)
Comment (jenniew) [reply pyspark/bigdl/models/inception/README.md]: done

File: pyspark/bigdl/models/inception/inception.py [hunk 3]
```diff
@@ -0,0 +1,316 @@
+from bigdl.nn.layer import *
+from optparse import OptionParser
+from bigdl.nn.criterion import *
+from bigdl.nn.initialization_method import *
+from bigdl.optim.optimizer import *
+from bigdl.transform.vision.image import *
+
+
+def t(input_t):
```
Comment (SeaOfOcean) [inline pyspark/bigdl/models/inception/inception.py]: why not use BigDL python table T ?

**Comment by zhichao-li** [comment]
is this PR ready to be merged? we are going to try the imagenet preprocess utility once it's ready.

File: pyspark/bigdl/models/inception/README.md [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+## Get the JAR
+You can build one by refer to the
+[Build Page](https://github.com/intel-analytics/BigDL/wiki/Build-Page) from the source code. We
+will release a pre-build package soon.
+
+## Prepare the data
+You can download imagenet-2012 data from <http://image-net.org/download-images>.
+ 
+After you download the files(**ILSVRC2012_img_train.tar** and **ILSVRC2012_img_val.tar**), 
+run the follow commands to prepare the data.
```
Comment (wzhongyuan) [inline pyspark/bigdl/models/inception/README.md]: follow -> following

File: pyspark/bigdl/models/inception/README.md [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+mkdir val
+mv ILSVRC2012_img_val.tar val/
+cd val
+tar -xvf ILSVRC2012_img_val.tar
+cat classes.lst | while read CLASS_NAME; do mkdir -p ${CLASS_NAME}; done
+cat img_class.lst | while read PARAM; do mv ${PARAM/ n[0-9]*/} ${PARAM/ILSVRC*JPEG /}; done
+rm ILSVRC2012_img_val.tar
+```
+
+Now all the images belong to the same category are moved to the same folder.
```
Comment (wzhongyuan) [inline pyspark/bigdl/models/inception/README.md]: belong -> belonging

File: pyspark/bigdl/models/inception/README.md [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+--checkpoint /models/inception
+
+```
+
+In the above commands
+* -f: where you put your ImageNet data, it should be a hdfs folder
+* --checkpoint: Where you cache the model/train_state snapshot. You should input a folder and
+make sure the folder is created when you run this example. The model snapshot will be named as
+model.#iteration_number, and train state will be named as state.#iteration_number. Note that if
+there are some files already exist in the folder, the old file will not be overwrite for the
```
Comment (wzhongyuan) [inline pyspark/bigdl/models/inception/README.md]: overwritten

File: pyspark/bigdl/models/inception/inception.py [hunk 7]
```diff
@@ -0,0 +1,356 @@
...
+                                ChannelNormalize(0.485, 0.456, 0.406, 0.229, 0.224, 0.225),
+                                MatToTensor(to_rgb=True),
+                                  ImageFrameToSample(input_keys=["imageTensor"], target_keys=["label"])
+                                ])
+    raw_val_data = get_inception_data(options.folder, sc, "val")
+    val_data = DataSet.image_frame(raw_val_data).transform(val_transformer)
+
+    # build model
+    if options.model != "":
+        inception_model = Model.load(options.model)
```
Comment (wzhongyuan) [inline pyspark/bigdl/models/inception/inception.py]: Uer Model.loadModel(...) to use bigdl format model ?

Comment (jenniew) [reply pyspark/bigdl/models/inception/inception.py]: Here we load model checkpoint, which is saved by model.save not model.saveModule. So use Model.load API.

File: pyspark/bigdl/models/inception/README.md (hunk 4 above)
Comment (jenniew) [reply pyspark/bigdl/models/inception/README.md]: updated

File: pyspark/bigdl/models/inception/README.md (hunk 5 above)
Comment (jenniew) [reply pyspark/bigdl/models/inception/README.md]: updated

File: pyspark/bigdl/models/inception/README.md (hunk 6 above)
Comment (jenniew) [reply pyspark/bigdl/models/inception/README.md]: updated

**Comment by jenniew** [comment]
Jenkins passed

**Comment by wzhongyuan** [comment]
LGTM

**Pull request merged**